zigbee: Improve zboss trace support in ncp sample (#5446)
nrfconnect/sdk-nrf: doc/nrf/libraries/zigbee/osif.rst

* :kconfig:`CONFIG_ZBOSS_OSIF_LOG_LEVEL` - Configures the custom logger options for the ZBOSS OSIF layer.

.. _zigbee_osif_zboss_osif_serial:

ZBOSS OSIF Serial
*****************

Setting the :kconfig:`CONFIG_ZIGBEE_HAVE_SERIAL` enables serial abstract for the ZBOSS OSIF.

> ZBOSS OSIF serial implements a set of functions used by ZBOSS stack for serial communication, from which two sets of functions with common purpose can be distinguish:

Author: We implements multiple functions, some of them are used for NCP communication, others are for logging traces.

* Handling NCP communication with HOST device - used only for :ref:`NCP architecture <ug_zigbee_platform_design_ncp>`
* Logging ZBOSS Traces - stack logs useful for debugging, received in binary format.

> .. _zigbee_osif_zigbee_async_serial:

Author: No header for this - internal use only

ZBOSS OSIF serial supports single backend for handling NCP communication - Zigbee Async Serial.

> It uses :ref:`Zephyr UART API<uart_api>` and can be configured to use UART peripheral or USB CDC ACM device.

Author: Using Kconfig option `CONFIG_ZIGBEE_UART_DEVICE_NAME` to select USB Serial.
This requires USB config to be enabled, NCP sample has it configured when built with `prj_usb.conf` conf file.

> Received data is internally buffered and if Logging ZBOSS Traces is supported, data to transmit is also buffered.

Author: There are three "backends" to choose from for logging ZBOSS traces and only one can be chosen.
To use Zigbee Async Serial for logging traces, we enable `ZBOSS_TRACE_BINARY_NCP_TRANSPORT_LOGGING`.

To configure this, use the following options:

* :kconfig:`CONFIG_ZIGBEE_HAVE_ASYNC_SERIAL` - to enable Zigbee Async Serial.
* :kconfig:`CONFIG_ZIGBEE_UART_DEVICE_NAME` - to specify serial device to use.
* :kconfig:`CONFIG_ZIGBEE_UART_SUPPORTS_FLOW_CONTROL` - set this option for serial devices that support flow control.
* :kconfig:`CONFIG_ZIGBEE_UART_RX_BUF_LEN` - to configure size of internal RX and TX buffer.

> * :kconfig:`ZBOSS_TRACE_BINARY_NCP_TRANSPORT_LOGGING` - to enable Logging ZBOSS Traces in binary format with Zigbee Async Serial.

Author: Should be, will do

> * :kconfig:`ZBOSS_TRACE_BINARY_NCP_TRANSPORT_LOGGING` - to enable Logging ZBOSS Traces in binary format with Zigbee Async Serial.

Author: will do

> Backend options for Logging ZBOSS Traces:

Reviewer: ```suggestion
Backend functions for logging ZBOSS traces
===========================

The backend functions for logging ZBOSS traces belong to the following groups:

> * Zigbee Async Serial - allows to use the same backend for NCP communication and Logging ZBOSS Traces. Logs ZBOSS Traces in binary format.
>   For how to configure, see :ref:`Zigbee Async Serial<zigbee_osif_zigbee_async_serial>`.

Reviewer: ```suggestion
* Zigbee Async Serial - This set of functions allows to use the same backend for NCP communication and logging ZBOSS traces.
When enabled, it logs ZBOSS traces in the binary format.
  For how to configure this backend, see `Zigbee async serial configuration options`_.
```
Althougn under the link provided there is barely any information about how to configure this option -- just one option is listed. If that's all, why not mention this option here? If more steps are needed, we need to expand that info.

Author: Will check this

* Zigbee Serial Logger - uses :ref:`Zephyr UART API<uart_api>` and can be configured to use UART peripheral or USB CDC ACM device.

>   Data is buffered internally in ring buffer and printed in binary format. Use the following options to configure it:

Reviewer: ```suggestion
* Zigbee serial logger - This set of functions uses Zephyr's :ref:`UART API <zephyr:uart_api>` and can be configured to use either the UART peripheral or the USB CDC ACM device.
  Data is buffered internally in ring buffer and printed in the binary format.
* Zigbee logger - This set of functions uses Zephyr's :ref:`zephyr:logging_api` API for logging hexdumps of received binary data.
  Data is buffered internally in ring buffer.

Zigbee serial logger configuration options
----------------------------------------

Use the following options to configure the Zigbee serial logger:

  * :kconfig:`CONFIG_ZBOSS_TRACE_BINARY_LOGGING` - to enable Logging ZBOSS Traces with Zigbee Serial Logger.
  * :kconfig:`CONFIG_ZBOSS_TRACE_UART_LOGGING` - UART serial backend.
  * :kconfig:`CONFIG_ZBOSS_TRACE_USB_CDC_LOGGING` - USB CDC ACM serial backend.
  * :kconfig:`CONFIG_ZBOSS_TRACE_LOGGER_DEVICE_NAME` - to specify the serial device to use.
  * :kconfig:`CONFIG_ZBOSS_TRACE_LOGGER_BUFFER_SIZE` - to specify size of internal ring buffer.

* Zigbee logger - uses Zephyr's :ref:`zephyr:logging_api` API for Logging hexdumps of received binary data.

>   Data is buffered internally in ring buffer. Use the following options to configure it:

Reviewer: Zigbee logger configuration options
-------------------------------------

Use the following options to configure the Zigbee logger:

Author: will do
